src/parser/jobs/war/modules/Gauge.tsx

```diff
@@ -51,46 +30,60 @@ export class Gauge extends CoreGauge {
 
 	@dependency private actors!: Actors
 	@dependency private cooldowns!: Cooldowns
+	@dependency private data!: Data
 	@dependency private suggestions!: Suggestions
 
 	private beastGauge = this.add(new CounterGauge({
 		chart: {label: 'Beast Gauge', color: JOBS.WARRIOR.colour},
 	}))
+	private beastGaugeModifiers = new Map<number, CounterGaugeModifier>([
+		// Builders
+		[this.data.actions.MAIM.id, {combo: 10}],
+		[this.data.actions.STORMS_EYE.id, {combo: 10}],
+		[this.data.actions.STORMS_PATH.id, {combo: 20}],
+		[this.data.actions.MYTHRIL_TEMPEST.id, {combo: 20}],
+		[this.data.actions.INFURIATE.id, {action: 50}],
+		// Spenders
+		[this.data.actions.FELL_CLEAVE.id, {action: -50}],
+		[this.data.actions.DECIMATE.id, {action: -50}],
+		[this.data.actions.UPHEAVAL.id, {action: -20}],
+		[this.data.actions.ONSLAUGHT.id, {action: -20}],
+		[this.data.actions.CHAOTIC_CYCLONE.id, {action: -50}],
+		[this.data.actions.INNER_CHAOS.id, {action: -50}],
+	])
 
-	protected override init() {
-		super.init()
+	override initialise() {
+		super.initialise()
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+		const beastActions = Array.from(this.beastGaugeModifiers.keys())
+		this.addEventHook(playerFilter.type('action').action(oneOf(beastActions)), this.onGaugeModifier)
+		this.addEventHook(playerFilter.type('combo').action(oneOf(beastActions)), this.onGaugeModifier)
+
+		const infuriateReducerIds = INFURIATE_REDUCERS.map(key => this.data.actions[key].id)
+		this.addEventHook(playerFilter.type('action').action(oneOf(infuriateReducerIds)), () => this.cooldowns.reduce('INFURIATE', INFURIATE_CDR))
 
-		this.addEventHook(
-			['combo', 'cast'],
-			{by: 'player', abilityId: Array.from(BEAST_MODIFIERS.keys())},
-			this.onGaugeModifier,
-		)
-		this.addEventHook(
-			'cast',
-			{by: 'player', abilityId: INFURIATE_REDUCERS},
-			() => {
-				this.cooldowns.reduce('INFURIATE', INFURIATE_CDR)
-			},
-		)
 		this.addEventHook('complete', this.onComplete)
 	}
 
-	private onGaugeModifier(event: LegacyComboEvent | CastEvent) {
-		const modifiers = BEAST_MODIFIERS.get(event.ability.guid) || {}
+	private onGaugeModifier(event: Events['action' | 'combo']) {
+		const modifiers = this.beastGaugeModifiers.get(event.action)
```

Comment:
This should be renamed to `modifier`, since it's always going to be a unique single object anyway.

Reply:
I'll sort this out later then.